.devcontainer.json

```diff
@@ -0,0 +1,39 @@
+{
+  "name": "terraform-provider-argocd",
+  // officiall MS template from https://github.com/devcontainers/templates/tree/main/src/go
+  "image": "mcr.microsoft.com/devcontainers/go:1.20-bookworm",
+  // see https://containers.dev/features for a comprehensive list of available features
+  "features": {
+    // https://github.com/devcontainers/features/tree/main/src/docker-in-docker
+    "ghcr.io/devcontainers/features/docker-in-docker:2": {
```

Comment:
Seems the better option than "Docker outside of Docker" to be able to run docker containers according to the description of both features.